gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -90,16 +100,26 @@
   private final LocationManager locationManager;
   private final FeatureRepository featureRepository;
 
+  private final List<Point> vertices = new ArrayList<>();
+
   @Hot private final Subject<Boolean> locationLockChangeRequests = PublishSubject.create();
   @Hot private final Subject<CameraUpdate> cameraUpdateSubject = PublishSubject.create();
 
+  /** Polyline drawn by the user but not yet saved as polygon. */
+  @Hot
+  private final PublishProcessor<ImmutableList<Point>> drawnPolylineVertices =
+      PublishProcessor.create();
+
   @Hot(replays = true)
   private final MutableLiveData<Integer> mapControlsVisibility = new MutableLiveData<>(VISIBLE);
 
   private final MutableLiveData<Boolean> completeButtonVisible = new MutableLiveData<>(false);
 
   private final MutableLiveData<Boolean> addPolygonVisible = new MutableLiveData<>(false);
 
+  @Hot(replays = true)
+  private final MutableLiveData<Boolean> addPolygonPoints = new MutableLiveData<>(false);
+
```

Comment:
This is not being used anywhere. Why is it defined?

Reply:
is used in layout and is responsible for visibility of AddPoint.

Reply:
If that's the case then the name is misleading - it implies the stream contains actual points to be added to the polygon. Shouldn't it be `addVertexButtonVisibility` or similar?

Reply:
renamed